src/apps/backend/modules/vendor_account/internal/vendor_account_reader.py

```diff
@@ -0,0 +1,16 @@
+from modules.vendor_account.errors import VendorAccountWithSameNameAndAccountExistsError
+from modules.vendor_account.internal.store.vendor_account_repository import VendorAccountRepository
+from modules.vendor_account.types import NameAlreadyExistParams, VendorAccount
+
+
+class VendorAccountReader:
+    @staticmethod
+    def check_name_not_exist(params: NameAlreadyExistParams) -> VendorAccount:
```

Comment:
Please move away from this pattern.

Consider exposing:

`get_venor_account_optionalaccount_id, vendor_account_name: str)`

The write method `create_vendor_account` can call this method and if this returns an instance, it can throw `VendorAccountExists` error. You can pass the `vendor_account_name` and `account` if needed and construct the message inside the error class itself.